### ISSUE TITLE
Bad exit code when uncaught exception

### ISSUE THREAD

**Issue by vvo** [scottcorgan/tap-spec#30]
Given this javascript file:

```js
dsadsadas;
```

```sh
node fail.js | tap-spec
# .. lots of errors
# then
# total:     0
#  passing:   0
#  duration:  4ms
echo $?
0
```

While the exit code should be -1

When does this happens? When your code fails before being able to start the test suite,
for example when your test-runner failed.

What do you think @scottcorgan?

**Comment by scottcorgan** [comment]
I've run across this issue before. It has to do with the underlying parser. I haven't found a good solution yet, but I'm working on something that should fix this for `4.0`, among other things.

Thoughts on a solution?

**Comment by vvo** [comment]
Great that you already knew it and working on a fix, I do not have any particular insight as I did not dig into the tap-spec projects.

I stopped using it even if I love it, it's only display, I count on you to solve this one ;)

Thanks.

**Comment by mynameistechno** [comment]
:+1:

**Comment by scottcorgan** [comment]
There's a discussion going on here about how to handle input that failed ... https://github.com/substack/tap-parser/issues/22

A fix didn't make it in't `4.0`, but I'm still thinking of the best way to do this.

**Comment by remy** [comment]
So, I've found an unobtrusive work around for this problem. It's kinda ugly, but it works.

In my `package.json`, I now have:

```json
  "scripts": {
    "test": "node test/*.test.js | tap-spec && node -e 'process.exit(process.argv[1])' ${PIPESTATUS[0]}"
  }
```

Now when there's a bug in the test harness, and something in `*.test.js` throws, although `tap-spec` can't see the exit code, my final check picks it up.

Like I said, not pretty, but it works.

**Comment by scottcorgan** [comment]
This is fixed and published as `4.1.0`. Thanks for the patience y'all!
